lib/SIL/SILVerifier.cpp

```diff
@@ -751,7 +751,8 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
 
     // If there are substitutions, verify them and apply them to the callee.
     if (subs.empty()) {
-      require(!fnTy->isPolymorphic(),
+      require(!fnTy->isPolymorphic() ||
+                  fnTy->getGenericSignature()->areAllParamsConcrete(),
```

Comment:
@slavapestov I'm not sure what you suggest here by "fixing it at the source". Do you suggest to use a non-polymorphic  SILFunctionType for such full specializations? Or do you suggest to fix the isPolymorphic test, so that it does not treat a type with a fully generic signature as polymorphic?